core/src/org/pentaho/di/core/row/RowMeta.java

```diff
@@ -129,7 +129,10 @@ public String toString() {
   public void setValueMetaList( List<ValueMetaInterface> valueMetaList ) {
     this.valueMetaList = valueMetaList;
     // we have new list and we need re-hash
-    indexValueMap();
+    valueIndexMap.clear();
+    for ( int i = 0; i < this.valueMetaList.size(); i++ ) {
+      valueIndexMap.put( this.valueMetaList.get( i ).getName().toLowerCase(), i );
```

Comment:
Not sure, but wouldn't hurt. BRB

Thanks,
-Nick Hudak


-------- Original Message --------
From: Matt Burgess <notifications@github.com>
Sent: Monday, January 19, 2015 05:15 PM
To: pentaho/pentaho-kettle <pentaho-kettle@noreply.github.com>
Subject: Re: [pentaho-kettle] [BACKLOG-1938]Lazy ValueMeta name caching (#950)
CC: Nick Hudak <nhudak@pentaho.com>


In core/src/org/pentaho/di/core/row/RowMeta.java<https://github.com/pentaho/pentaho-kettle/pull/950#discussion-diff-23190510>:

> @@ -129,7 +129,10 @@ public String toString() {
>    public void setValueMetaList( List<ValueMetaInterface> valueMetaList ) {
>      this.valueMetaList = valueMetaList;
>      // we have new list and we need re-hash
> -    indexValueMap();
> +    valueIndexMap.clear();
> +    for ( int i = 0; i < this.valueMetaList.size(); i++ ) {
> +      valueIndexMap.put( this.valueMetaList.get( i ).getName().toLowerCase(), i );


Does the map need to be a ConcurrentHashMap?

-
Reply to this email directly or view it on GitHub<https://github.com/pentaho/pentaho-kettle/pull/950/files#r23190510>.